server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/LongValuesComparatorSource.java

```diff
@@ -120,7 +120,13 @@ public FieldComparator<?> newComparator(String fieldname, int numHits, boolean e
         final long lMissingValue = (Long) missingObject(missingValue, reversed);
         // NOTE: it's important to pass null as a missing value in the constructor so that
         // the comparator doesn't check docsWithField since we replace missing values in select()
-        return new LongComparator(numHits, null, null, reversed, false) {
+        return new LongComparator(
+            numHits,
+            fieldname,
+            null,
+            reversed,
+            indexFieldData.enableSortOptimizationForCustomComparators(IndexNumericFieldData.NumericType.LONG)
```

Comment:
Do you think we could move it to constructor instead? this is not changing from call to call

Reply:
Yes. I acutlly moved ahead and made it to constructor but I realzied, we need `enableSkipping` functionaltiy to be true by default and disable case basis. So I introduced `enableSkipping` in parent class and have set it to `true` by default. With API to disable it when we do casting in sort field. I think this is clean approach, what you think ?

Reply:
> Yes. I acutlly moved ahead and made it to constructor but I realzied, we need enableSkipping functionaltiy to be true by default and disable case basis.

I like they idea but the `XFieldComparatorSource` becomes mutable now, if you just pass `enableSkipping` in constructor, it would make it immutable.

Reply:
Right. The problem is we will have to change at many places including `dateNanosComparatorSource` & `dateComparatorSource` methods implementations if we include this in constructior. And top of it, all `XFieldComparatorSource` implementation for different numeric types, that will lead to several other changes in test files too (we can surely avoid those but it will require to create one more constructor in those sources implementation).

I just feel, this setting to be kind of `TODO` untill we add support for skipping logic in case of casting of sort field too.
And so current implementation is inline with [SortField::SetOptimizeSortWIthPoints](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/search/SortField.java#L686), where SortField is enabled by default with optimization but can be disable case basis. 

I would kind of prefer to avoid big changes in constructor for this. If we decide to support this enablSKipping for caste base in future, it will be just couple of files change that time too.